kernel/sched.c

```diff
@@ -81,30 +81,50 @@ static inline bool is_thread_dummy(struct k_thread *thread)
 }
 #endif
 
-bool z_is_t1_higher_prio_than_t2(struct k_thread *thread_1,
-				 struct k_thread *thread_2)
+/*
+ * Return value same as e.g. memcmp
+ * > 0 -> thread 1 priority  > thread 2 priority
+ * = 0 -> thread 1 priority == thread 2 priority
+ * < 0 -> thread 1 priority  < thread 2 priority
+ * Do not rely on the actual value returned aside from the above.
+ * (Again, like memcmp.)
+ */
+int32_t z_cmp_t1_prio_with_t2(struct k_thread *thread_1,
+	struct k_thread *thread_2)
 {
-	if (thread_1->base.prio < thread_2->base.prio) {
-		return true;
+	/* `prio` is <32b, so the below cannot overflow. */
+	int32_t b1 = thread_1->base.prio;
+	int32_t b2 = thread_2->base.prio;
+
+	if (b1 != b2) {
+		/* This is deliberately backwards - lower `prio`
+		 * means higher effective priority.
```

Comment:
I'd suggest removing the comment.  Inverted priority sense is a common enough paradigm that anyone working on this code should be expected to understand it, and the value here matches the convention we use in the public API.

Reply:
Fair enough, shall do.

- [ ] Remove trivial comment.